apps/chat/src/components/Common/ApplicationDialog.tsx

```diff
@@ -36,289 +30,121 @@ import Modal from '@/src/components/Common/Modal';
-  const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
-  const [isPublishing, setIsPublishing] = useState(false);
-
-  const entity = selectedApplication && {
-    name: selectedApplication.name,
-    id: ApiUtils.decodeApiUrl(
-      selectedApplication.id ||
-        (selectedApplication as unknown as { application: string }).application,
-    ),
-    folderId: getFolderIdFromEntityId(selectedApplication.name),
-  };
-
-  const resetForm = () => {
-    setName('');
-    setVersion('');
-    setDescription('');
-    setFeatures('');
-    setMaxAttachments(0);
-    setCompletionUrl('');
-    setLocalLogoFile(undefined);
-    setDeleteLogo(false);
-    setFilterParams([]);
-  };
-
-  const loading = useAppSelector((state) => state.application.loading);
-
-  const handleChangeFilterParams = useCallback((filterParams: string[]) => {
-    setFilterParams(filterParams);
-  }, []);
+  selectedApplication,
+}) => {
+  const {
+    register,
+    handleSubmit,
+    setValue,
+    clearErrors,
+    trigger,
+    control,
+    formState: { errors, isValid },
+  } = useForm<FormData>({
+    mode: 'onChange',
+    reValidateMode: 'onChange',
+  });
 
-  const handleClose = useCallback(() => {
-    setSubmitted(false);
-    resetForm();
-    onClose(false);
-  }, [onClose]);
+  const { t } = useTranslation(Translation.Chat);
 
-  const nameOnBlurHandler = (e: FocusEvent<HTMLInputElement>) => {
-    const newName = prepareEntityName(e.target.value.trim(), {
-      forRenaming: true,
-    });
-    const pattern = /^[^!@#$^*()+]{2,160}$/;
-
-    if (!pattern.test(newName)) {
-      setNameError(
-        t(
-          'Name should be 2 to 160 characters long and should not contain special characters',
-        ),
-      );
-    } else {
-      setNameError(null);
-    }
-    setName(newName);
-    onBlur(e);
-  };
+  const dispatch = useAppDispatch();
 
-  const nameOnChangeHandler = (e: ChangeEvent<HTMLInputElement>) => {
-    const newName = e.target.value;
-    setName(newName);
+  const files = useAppSelector(FilesSelectors.selectFiles);
 
-    if (notAllowedSymbolsRegex.test(newName)) {
-      setNameError(t("You can't type special characters"));
-    } else {
-      setName(newName);
-      setNameError(null);
-    }
-  };
+  const [deleteLogo, setDeleteLogo] = useState(false);
+  const [localLogoFile, setLocalLogoFile] = useState<string | undefined>();
+  const [inputAttachmentTypes, setInputAttachmentTypes] = useState<string[]>(
+    [],
+  );
 
-  const versionOnChangeHandler = (e: ChangeEvent<HTMLInputElement>) => {
-    const newVersion = e.target.value.trim();
-    const partialPattern = /^[0-9]+(\.[0-9]*)*$/;
-    if (partialPattern.test(newVersion) || newVersion === '') {
-      setVersion(newVersion);
-      setVersionError(null);
-    }
-  };
+  const [featuresInput, setFeaturesInput] = useState(
+    selectedApplication &&
+      selectedApplication.features &&
+      safeStringify(selectedApplication?.features),
```

Comment:
Check if value exists into `safeStringify`
```suggestion
      safeStringify(selectedApplication?.features),
```